File: SaveTheCat.Frontend/src/types/beatSheet.ts
export interface BeatSheetDto {
    id: string;
    projectId: string;
    title: string;
    logline: string;
    genre: string;
    date: string;
    openingImage: string;
    themeStated: string;
    setUp: string;
    catalyst: string;
    debate: string;
    breakIntoTwo: string;
    bStory: string;
    funAndGames: string;
    midpoint: string;
    badGuysCloseIn: string;
    allIsLost: string;
    darkNightOfTheSoul: string;
    breakIntoThree: string;
    finale: string;
    finalImage: string;
}

export type UpdateBeatSheetDto = Omit<BeatSheetDto, 'id' | 'projectId' | 'title' | 'date'>;

File: SaveTheCat.Backend/SaveTheCat.Domain/Entities/BeatSheet.cs
namespace SaveTheCat.Domain.Entities;

public class BeatSheet
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public required string Logline { get; set; }
    public required string Genre { get; set; }
    public required string Date { get; set; } 
    public required string OpeningImage { get; set; }
    public required string ThemeStated { get; set; }
    public required string SetUp { get; set; }
    public required string Catalyst { get; set; }
    public required string Debate { get; set; }
    public required string BreakIntoTwo { get; set; }
    public required string BStory { get; set; }
    public required string FunAndGames { get; set; }
    public required string Midpoint { get; set; }
    public required string BadGuysCloseIn { get; set; }
    public required string AllIsLost { get; set; }
    public required string DarkNightOfTheSoul { get; set; }
    public required string BreakIntoThree { get; set; }
    public required string Finale { get; set; }
    public required string FinalImage { get; set; }

    // Relación 1-a-1 con Proyecto
    public Guid ProjectId { get; set; }
    public virtual Project? Project { get; set; }
}
